### PR TITLE
chore: Initial edits to CODEOWNERS

### LINKED ISSUE #3665: Review & update CODEOWNERS
We need to review the `CODEOWNERS` file to ensure it's accurate and fair. Given the recent additions to the team, it's a good opportunity to think about domain ownership.

- [ ] Remove `a-rodin` since he is no longer on the team.
- [ ] Remove `LucioFranco` since he is no longer on the team.
- [ ] Thank about distributing load more fairly.
- [ ] Ensure that all sources, transforms, sinks, and other `/src` files have an owner assigned.
- [ ] Ensure that the `.github/workflows./*` files are assigned.
- [ ] Ensure that James and myself review changes to important root-level files `PRIVACY.md`, `LICENSE`, `CONTRIBUTING`, `CODE_OF_CONDUCT`, etc.

### PR DESCRIPTION
**Pull request by jamtur01**
Closes #3665

### CODE REVIEW

File: .github/CODEOWNERS
Comment (fanatid) [inline .github/CODEOWNERS:45]: I removed this file in https://github.com/timberio/vector/pull/3627 but forgot to change CODEOWNER :disappointed: can we fix it in this PR?

File: .github/CODEOWNERS
Comment (bruceg) [inline .github/CODEOWNERS:37]: Do we need to list sources that have no owner?

File: .github/CODEOWNERS
Comment (jamtur01) [reply .github/CODEOWNERS:45]: runtime.rs or region.rs or both?

Comment (jamtur01) [reply .github/CODEOWNERS:45]: nm - found it.

Comment (fanatid) [reply .github/CODEOWNERS:45]: Only `runtime.rs`

File: .github/CODEOWNERS
Comment (bruceg) [inline .github/CODEOWNERS:44]: I should maybe own this one, as my name is on most of `git blame`.

File: .github/CODEOWNERS
Comment (fanatid) [reply .github/CODEOWNERS]: I was late with reply on 4 seconds :laughing: thanks @jamtur01

File: .github/CODEOWNERS
Comment (bruceg) [inline .github/CODEOWNERS:46]: This should be `/src/tls` as it is now a directory.

**Comment by binarylogic** [comment]
@timberio/vector feel free to make edits directly if it is clear you should be an owner.

File: .github/CODEOWNERS
Comment (Hoverbear) [inline .github/CODEOWNERS:28]: I don't actually know anything about these scripts! (Also you mispelled it)

**Comment by Hoverbear** [comment]
@jamtur01 If you get a chance feel free to add me as owner to `tests/data/wasm-tests`, `lib/vector-wasm` and `src/wasm` :)

**Review by MOZGIII — APPROVED** [review]

File: .github/CODEOWNERS
Comment (jamtur01) [reply .github/CODEOWNERS]: I left them in so people could see them and claim them. I'll remove any we decide not to claim.

**Comment by leebenson** [comment]
I wouldn't mind getting in on any API action eventually, when #3645 / #3514 settle and it's a bit clearer what's going on there. We can edit at that time. For now, I'm a tourist-- I think it's more for others to mark their turf.

File: .github/CODEOWNERS
Comment (MOZGIII) [inline .github/CODEOWNERS:10]:
```suggestion
.github/workflows/ @jamtur01
```

Comment (MOZGIII) [reply .github/CODEOWNERS:10]: We probably should move this section to the bottom alongside with `/.github/workflows/e2e.yml`

**Comment by MOZGIII** [comment]
It looks like the pattern we used until now is starting paths with `/`. This PR adds some entries that don't follow that pattern. I don't mind switching to bare paths - the semantics is different but it's not distinctive in this case - but I think consistency is important, and we should use one form everywhere.

**Review by fanatid — APPROVED** [review]

**Comment by JeanMertz** [comment]
I'm happy to be assigned to the following domains:

* `ansi_stripper` transform
* internal events (although @lukesteensen is still far more knowledgeable on this domain, but he probably has enough on his plate already)
* the new `remap` transform, once it lands
* anything Kubernetes related, if it helps

I'm also happy to be assigned to any other areas that could use some extra eyeballs. It'll just take me a couple of days to get up to speed on them.

**Review by Hoverbear — APPROVED** [review]

**Pull request merged**
